pkg/providers/vault/tests/main.go

```diff
@@ -0,0 +1,169 @@
+	var createTokenResp creatTokenResponse
+
+	if err := json.Unmarshal(response, &createTokenResp); err != nil {
+		panic(err)
+	}
+
+	authToken := createTokenResp.Auth.ClientToken
+
+	logger.Debug(fmt.Sprintf("Created new client token: %v", authToken))
+
+	config := vault.SecretConfig{
+		Protocol:       "http",
+		Host:           vaultHost,
+		Port:           vaultPort,
+		Authentication: vault.AuthenticationInfo{AuthToken: authToken}}
+
+	ctx := context.Background()
+	errChan := make(chan error)
+
+	// Create SecrectClient to start the token refresh cycle
+	_, err := vault.NewSecretClient(config, logger, ctx, errChan)
+	if err != nil {
+		panic(err)
+	}
+
+	timesUpticker := time.NewTicker(totalTimeWindow)
+	lookUpSelfTicker := time.NewTicker(1 * time.Second)
+	failures := 0
+	tokenRefreshes := 0
+	oldTTL := 0
+
+	// Check for errors on error channel and query lookup for ttl changes
+	for {
+		select {
+		case err = <-errChan:
+			logger.Error(fmt.Sprint("Error from errChan ", err))
+			failures++
+
+		case <-lookUpSelfTicker.C:
+			var response lookUpSelfResponse
+			lookupData := makeHTTPCall(urlLookupSelf, "GET", "", authToken)
+
+			if err := json.Unmarshal(lookupData, &response); err != nil {
+				panic(err)
+			}
+
+			logger.Debug(fmt.Sprintf("Check lookup-self for token ttl: %v, period: %v", response.Data.TTL, response.Data.Period))
+
+			// give a one second window to allow for token refresh time
+			if response.Data.TTL < ((response.Data.Period / 2) - 1) {
+				failures++
+				logger.Debug(fmt.Sprintf("TTL fell below half of period ttl is %d, period is %d", response.Data.TTL, response.Data.Period))
+			}
+
+			if response.Data.TTL > oldTTL && oldTTL != 0 {
+				tokenRefreshes++
+			}
+
+			oldTTL = response.Data.TTL
+
+		case <-timesUpticker.C:
+			logger.Debug("Time is up!")
+			timesUpticker.Stop()
+			lookUpSelfTicker.Stop()
+			if failures > 0 {
+				logger.Debug(fmt.Sprintf("Report: Token failed to refresh at least once, check the logs above for more details."))
+			} else {
+				logger.Debug(fmt.Sprintf("Report: Token refreshed as expected with a total time window of %s, and period of %s there were %v token refreshes.", totalTimeWindow.String(), tokenPeriod.String(), tokenRefreshes))
+			}
+			return
+		}
+	}
+}
+
+func makeHTTPCall(url string, httpMethod string, data string, token string) []byte {
+
+	req, err := http.NewRequest(httpMethod, url, bytes.NewBuffer([]byte(data)))
+	req.Header.Set("X-Vault-Token", token)
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return body
+}
+
+type auth struct {
+	ClientToken   string `json:"client_token"`
+	LeaseDuration int    `json:"lease_duration"`
+	Renewable     bool   `json:"renewable"`
+}
+
+type creatTokenResponse struct {
```

Comment:
Prefer to use complete word of createTokenResponse VS. creatTokenResponse.

Reply:
ok